tests/_test_msui/test_msui.py

```diff
@@ -328,3 +328,225 @@ def test_flight_track_io(self, mockload, mocksave, mockq, mocki, mockw):
         assert self.window.listFlightTracks.count() == 2
         assert os.path.exists(self.save_ftml)
         os.remove(self.save_ftml)
+
+class Test_MSUIMainWindow:
+    def setUp(self):
+        # Create a QApplication instance for Qt tests
+        self.app = QtWidgets.QApplication([])
+
+        # Create temporary directory for JSON files
+        self.temp_dir = tempfile.mkdtemp()
+
+        # Mock config_loader to return temporary directory
+        self.config_patcher = patch("mslib.msui.msui_mainwindow.config_loader")
+        self.mock_config = self.config_patcher.start()
+        self.mock_config.side_effect = lambda dataset=None, default=False: {
+            "data_dir": self.temp_dir,
+            "new_flighttrack_template": [("START", 0, 0), ("END", 10, 10)],
+            "new_flighttrack_flightlevel": 350,
+            "restore_views": True,
+            "filepicker_default": "qt"
+        }.get(dataset, {})
```

Comment:
A mock shouldn't be necessary to set the relevant configurations, there is `modify_config_file` for this that other tests use already.